### PR TITLE
{lang,lib}[GCC/10.2.0,fosscuda/2020b] Boost.Python v1.74.0, PyCUDA v2020.1 w/ Python 2.7.18 + 3.8.6

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi8029 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/cb9623011c2ecdc127e782b5ad2e1dcf for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
**FAILED**
Build succeeded for 52 out of 64 (2 easyconfigs in total)
taurusml25 - Linux RHEL 7.6, POWER, 8335-GTX (power9le), Python 2.7.5
See https://gist.github.com/b76fbc0ba0bfd39109de46167dfd5e10 for a full test report.

File: easybuild/easyconfigs/p/PyCUDA/PyCUDA-2020.1-fosscuda-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+local_preinstallopts = "./configure.py --cuda-root=$EBROOTCUDA --boost-inc-dir=$EBROOTBOOST/include/boost "
+local_preinstallopts += "--boost-lib-dir=$EBROOTBOOST/lib --no-use-shipped-boost "
+local_preinstallopts += "--boost-python-libname=boost_python38 && "
+
+sanity_pip_check = True
+use_pip = True
+
+exts_default_options = {
+    'source_tmpl': '%(namelower)s-%(version)s.tar.gz',
+    'source_urls': ['https://pypi.python.org/packages/source/%(nameletterlower)s/%(namelower)s'],
```
Comment (akesandgren) [inline easybuild/easyconfigs/p/PyCUDA/PyCUDA-2020.1-fosscuda-2020b.eb:27]: Any reason you're not using SOURCELOWER_TAR_GZ and PYPI_LOWER_SOURCE here?

**Comment by akesandgren** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/p/PyCUDA/PyCUDA-2020.1-fosscuda-2020b.eb (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/p/PyCUDA/PyCUDA-2020.1-fosscuda-2020b.eb:27]: I'm wondering why that isn't the default for python packages already...

Comment (akesandgren) [reply easybuild/easyconfigs/p/PyCUDA/PyCUDA-2020.1-fosscuda-2020b.eb]: because it is just as common to  have PYPI_SOURCE or SOURCE_TAR_GZ ?

**Comment by Flamefire** [comment]
Test report by @Flamefire
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
taurusi8013 - Linux centos linux 7.9.2009, x86_64, AMD EPYC 7352 24-Core Processor (zen2), Python 2.7.5
See https://gist.github.com/eb35e8d6b0e96d052d8e4ec0ed93e283 for a full test report.

**Comment by akesandgren** [comment]
@boegelbot please test @ generoso

**Review by boegel — DISMISSED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3301.joltik.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6242 CPU @ 2.80GHz (cascadelake), Python 3.6.8
See https://gist.github.com/8d267cf42836bc2b1b5c47b284cf2258 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2700.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/056fe0cd0ef5ec5518e9ffd43958c959 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3507.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/ca6ff94a53ae5a89b99aa04a9bb2ce88 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2700.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/9a52a552ac0527181fd7353d74cda75b for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @Flamefire!

**Pull request merged**